hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -1256,68 +1257,69 @@ vlapic_icrlo_write_handler(struct acrn_vlapic *vlapic)
 	if ((mode == APIC_DELMODE_FIXED) && (vec < 16U)) {
 		vlapic_set_error(vlapic, APIC_ESR_SEND_ILLEGAL_VECTOR);
 		dev_dbg(ACRN_DBG_LAPIC, "Ignoring invalid IPI %u", vec);
-		return 0;
-	}
-
-	dev_dbg(ACRN_DBG_LAPIC,
-		"icrlo 0x%08x icrhi 0x%08x triggered ipi %u",
-			icr_low, icr_high, vec);
+	} else if (((shorthand == APIC_DEST_SELF) || (shorthand == APIC_DEST_ALLISELF))
+			&& ((mode == APIC_DELMODE_NMI) || (mode == APIC_DELMODE_INIT)
+			|| (mode == APIC_DELMODE_STARTUP))) {
+			dev_dbg(ACRN_DBG_LAPIC, "Invalid ICR value");
+	} else {
 
-	if (((shorthand == APIC_DEST_SELF) || (shorthand == APIC_DEST_ALLISELF))
-		&& ((mode == APIC_DELMODE_NMI) || (mode == APIC_DELMODE_INIT)
-		|| (mode == APIC_DELMODE_STARTUP))) {
-		dev_dbg(ACRN_DBG_LAPIC, "Invalid ICR value");
-		return 0;
-	}
+		dev_dbg(ACRN_DBG_LAPIC,
+			"icrlo 0x%08x icrhi 0x%08x triggered ipi %u",
+				icr_low, icr_high, vec);
 
-	switch (shorthand) {
-	case APIC_DEST_DESTFLD:
-		vlapic_calcdest(vlapic->vm, &dmask, dest, phys, false);
-		break;
-	case APIC_DEST_SELF:
-		bitmap_set_lock(vlapic->vcpu->vcpu_id, &dmask);
-		break;
-	case APIC_DEST_ALLISELF:
-		dmask = vm_active_cpus(vlapic->vm);
-		break;
-	case APIC_DEST_ALLESELF:
-		dmask = vm_active_cpus(vlapic->vm);
-		bitmap_clear_lock(vlapic->vcpu->vcpu_id, &dmask);
-		break;
-	default:
-		/*
-		 * All possible values of 'shorthand' has been handled in prior
-		 * case clauses.
-		 */
-		break;
-	}
+		switch (shorthand) {
+		case APIC_DEST_DESTFLD:
+			vlapic_calcdest(vlapic->vm, &dmask, dest, phys, false);
+			break;
+		case APIC_DEST_SELF:
+			bitmap_set_lock(vlapic->vcpu->vcpu_id, &dmask);
+			break;
+		case APIC_DEST_ALLISELF:
+			dmask = vm_active_cpus(vlapic->vm);
+			break;
+		case APIC_DEST_ALLESELF:
+			dmask = vm_active_cpus(vlapic->vm);
+			bitmap_clear_lock(vlapic->vcpu->vcpu_id, &dmask);
+			break;
+		default:
+			/*
+			 * All possible values of 'shorthand' has been handled in prior
+			 * case clauses.
+			 */
+			break;
+		}
 
-	for (vcpu_id = 0U; vcpu_id < vlapic->vm->hw.created_vcpus; vcpu_id++) {
-		if ((dmask & (1UL << vcpu_id)) != 0UL) {
-			target_vcpu = vcpu_from_vid(vlapic->vm, vcpu_id);
+		for (vcpu_id = 0U; vcpu_id < vlapic->vm->hw.created_vcpus; vcpu_id++) {
+			if ((dmask & (1UL << vcpu_id)) != 0UL) {
+				target_vcpu = vcpu_from_vid(vlapic->vm, vcpu_id);
```

Comment:
My previous comment is not addressed

Reply:
Just give it a try, remove it (the checking for target_vcpu == NULL),  MISRA doesn't report violation any more.   so,  I will remove it because when  vcpu_id < vlapic->vm->hw.created_vcpus,  the returned target_vcpu won't be NULL.